### PR TITLE
Update Accordion tests to use best testing practices

### PR DESCRIPTION
**Pull request by jcfilben**
#### What does this PR do?
Updated tests for the component to follow the testing guidelines//best practices that are being added to the contributing guide. We can use the Accordion tests as a reference for people to look at.

I also added `eslint-plugin-testing-library` to catch errors related to using react testing library. For now I only enabled it for the Accordion test file.

Note: switching from `container.firstChild` to `asFragment()` is causing the majority of snapshot changes

#### Where should the reviewer start?
Accordion-test.tsx

#### What are the relevant issues?
Related to grommet/hpe-design-system#1959

#### Do the grommet docs need to be updated?
No

#### Should this PR be mentioned in the release notes?
No

#### Is this change backwards compatible or is it a breaking change?
Backwards compatible

### CODE REVIEW

**Review by halocline — APPROVED** [review]

File: src/js/components/Accordion/__tests__/Accordion-test.tsx [hunk 1]
```diff
@@ -26,20 +27,20 @@ describe('Accordion', () => {
 
     const results = await axe(container);
     expect(results).toHaveNoViolations();
-    expect(container).toMatchSnapshot();
+    expect(asFragment()).toMatchSnapshot();
   });
 
   test('no AccordionPanel', () => {
-    const { container } = render(
+    const { asFragment } = render(
```
Comment (taysea) [inline src/js/components/Accordion/__tests__/Accordion-test.tsx:34]: can `asFragment` come from "screen"? Or it needs to get defined in this way?

Comment (jcfilben) [reply src/js/components/Accordion/__tests__/Accordion-test.tsx:34]: Only queries (`getByRole`, `getByLabelText`, etc.) can come from screen so `asFragment` can't come from screen and has to be defined like that

Comment (taysea) [reply src/js/components/Accordion/__tests__/Accordion-test.tsx:34]: Thanks!

**Review by taysea — APPROVED** [review]
Thanks!

**Review by ericsoderberghp — APPROVED** [review]

**Pull request merged**